rfd/0007-rbac-oss.md

```diff
@@ -0,0 +1,160 @@
+---
+authors: Alexander Klizhentas (sasha@gravitational.com)
+state: discussion
+---
+
+# RFD 2 - Open Source Roles
+
+## What
+
+Enable role based access control in open source.
+
+## Why
+
+Provide better user experience for open source users.
+Unify user experience, testing and development for open source branches.
+
+## Details
+
+### New Open Source Features
+
+The following features will become available starting 5.1 version.
+
+**Role Based Access Control**
+
+All RBAC features, except FedRamp feature flags.
+
+**Approval Plugins**
+
+Some approval plugins will become available in the open source:
+
+* Approval Workflows Golang SDK and API
+* Webhook
+* Slack
+* Gitlab
+* Mattermost
+
+### Enteprise Only Features
+
+The following features will remain enterprise only.
+
+**FedRamp**
+
+* Max connections AC control (and all future AC controls):
+
+```yaml
+role:
+   options:
+     max_connections: 3
+```
+
+* Teleport fips mode flag
+
+```bash
+$ teleport start --fips
+```
+
+**Single Sign On**
+
+OIDC and SAML connectors
+
+**Approval Plugins**
+
+Some plugins will be require license to start
+and will be moved to private source repository.
```

Comment:
This will require some different CI/release practices but it shouldn't be too much of an issue. We might need to change the plugin naming scheme to make it obvious which plugins are OSS and which are not, though.